### DISCUSSION TITLE
深入了解kubebuilder - Qinng

### DISCUSSION THREAD

**Discussion by giscus** [qingwave/qingwave.github.io#76] [Announcements]
# 深入了解kubebuilder - Qinng

前文快速实现一个Kubernetes Operator介绍了kubebuilder工具，快速实现了一个Operator。今天我们深入水下，探寻kubebuilder究竟是如何工作的。

https://qingwave.github.io/k8s-kubebuilder-deep-dive/

**Comment by lysShub** [comment]
请问一下，`runWorker`  之下的逻辑都在循环调用，相当于`runWorker`在不停地被调用。这不就是死循环代码吗？是不是有什么同步阻塞策略或者异步阻塞原语之类的东西？

**Reply by lysShub** [reply]
抱歉，没有仔细看；常规模式有ratelimit，operator有next run duration
